File: src/main/java/uknowlab/designpattern4java/factory/ChicagoPizzaStore.java
package uknowlab.designpattern4java.factory;

import uknowlab.designpattern4java.factory.pizza.ChicagoCheesePizza;
import uknowlab.designpattern4java.factory.pizza.ChicagoPepperoniPizza;
import uknowlab.designpattern4java.factory.pizza.ChicagoVeggiePizza;
import uknowlab.designpattern4java.factory.pizza.Pizza;

public class ChicagoPizzaStore extends PizzaStore {

    @Override
    protected Pizza createPizza(String type) {
        return switch (type) {
            case "cheese" -> new ChicagoCheesePizza();
            case "pepperoni" -> new ChicagoPepperoniPizza();
            case "veggie" -> new ChicagoVeggiePizza();
            default -> null;
        };
    }
}

File: src/main/java/uknowlab/designpattern4kotlin/factory/ChicagoPizzaStore.kt
package uknowlab.designpattern4kotlin.factory

import uknowlab.designpattern4kotlin.factory.pizza.ChicagoCheesePizza
import uknowlab.designpattern4kotlin.factory.pizza.ChicagoPepperoniPizza
import uknowlab.designpattern4kotlin.factory.pizza.ChicagoVeggiePizza
import uknowlab.designpattern4kotlin.factory.pizza.Pizza

class ChicagoPizzaStore : PizzaStore() {
    override fun createPizza(type: String?): Pizza {
        return when (type) {
            "cheese" -> ChicagoCheesePizza()
            "pepperoni" -> ChicagoPepperoniPizza()
            "veggie" -> ChicagoVeggiePizza()
            else -> throw IllegalArgumentException("No such pizza.")
        }
    }
}
